pkg/sentry/arch/arch_aarch64.go

```diff
@@ -279,10 +288,37 @@ func (s *State) FullRestore() bool {
 	return false
 }
 
+// getSigreturnSymbolValueFromELF returns the symbol value of rt_sigreturn in vdso.so.
+func getSigreturnSymbolValueFromELF(contents []byte) uint64 {
```

Comment:
Rather than embedding and parsing the VDSO in a second package, it seems like this would be better off done in pkg/sentry/loader. Perhaps as part of the return value to loader.PrepareVDSO?

Reply:
I have tried it before.
If I moved the similar function into loader, such as loader.GetSymbolValueFromVDSO, then I can't call it (loader.GetSymbolValueFromVDSO("rt_sigreturn"))in arch module, because there is a "cycle in dependency graph error" issue.
I need more time to investigate it.
Thanks.

Reply:
How about this new version of the PR?
I can't call the loader.GetSymbolValueFromVDSO("rt_sigreturn") in arch module directly, because there is 'cycle in dependency' issue.